kmip/core/messages/payloads/mac.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2017 Pure Storage, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from kmip.core import attributes
+from kmip.core import enums
+from kmip.core.enums import Tags
+
+from kmip.core.objects import Data, MACData
+
+from kmip.core.primitives import Struct
+
+from kmip.core.utils import BytearrayStream
+
+
+# 4.33
+class MACRequestPayload(Struct):
+
+    def __init__(self,
+                 unique_identifier=None,
+                 cryptographic_parameters=None,
+                 data=None):
+        super(MACRequestPayload, self).__init__(
+            tag=enums.Tags.REQUEST_PAYLOAD)
+        self.unique_identifier = unique_identifier
+        self.cryptographic_parameters = cryptographic_parameters
+        self.data = data
```

Comment:
It is inconsistent, but there's little risk here. The clients can either handle these payloads differently or multiple types can be handled in the property code itself, whichever we prefer. The benefit here to using properties is that it simplifies the use of the payloads long term, merging how a caller sees the `unique_identifier` as a string with its internal representation as a serializable structure.

Moreover, now there's no risk of missing or forgetting a `validate` call, since all attribute access is implicitly type-checked. For example, if a caller instantiated the payload, and then modified the attributes directly, it is very likely they would not call `validate` once they were done to catch mistakes. Now, they know instantly if they made a mistake, any time they access one of these attributes.

As for debug-only builds, we really don't have any structure in place to do that in the library. Even if we did, we would still want to verify attribute types since user/client code could easily pass in bad value types and trigger lower-level errors (or worse, generate a "valid" serializable request that actually doesn't do what the user/client thinks it should do). If performance is really a concern for someone's application, they shouldn't be using Python.